poi-ooxml/src/main/java/org/apache/poi/xssf/usermodel/XSSFWorkbook.java

```diff
@@ -1269,6 +1272,19 @@ public String getSheetName(int sheetIx) {
         return sheetIterator();
     }
 
+    /**
+     * Returns a spliterator of the sheets in the workbook
+     * in sheet order. Includes hidden and very hidden sheets.
+     *
+     * @return a spliterator of the sheets.
+     *
+     * @since POI 5.2.0
+     */
+    @Override
+    public Spliterator<Sheet> spliterator() {
+        return new SheetSpliterator();
```

Comment:
Is this better than `Spliterators.spliterator(sheets, Spliterator.ORDERED)`?

Reply:
Its actually the same, because `SheetSpliterator` is delegating to `sheets.spliterator()`, which is actually calling `Spliterators.spliterator(this, Spliterator.ORDERED)`.

I added `SheetSpliterator` to be consistent with `sheetIterator()`, which creates an instance of `SheetIterator` to delegate `sheets.iterator()`. But now that I think about it, its probably better to simply do a cast, reducing an object instantiation, eg.
```java
@Override
@SuppressWarnings("unchecked")
public Spliterator<Sheet> spliterator() {
    return (Spliterator<Sheet>)(Spliterator<? extends Sheet>) sheets.spliterator();
}
```

Reply:
I don't think you'll need to cast.

Reply:
Unfortunately `sheets.spliterator()` returns `Spliterator<XSSFSheet>`

Reply:
can you try omitting the cast and see if compiles? I just added spliterators in another project and was suprised that I didn't need to cast but it worked with the cast - https://github.com/pjfanning/excel-streaming-reader/commit/4ac5fb854498bf18a900c5ef95da6f07a37a7570

Reply:
I tried, it doesn't work. Using `Spliterators.spliterator(this, Spliterator.ORDERED)` avoids having to cast because [Spliterators.spliterator(Collection, int)](https://docs.oracle.com/javase/8/docs/api/java/util/Spliterators.html#spliterator-java.util.Collection-int-) takes in `Collection<? extends T>` and returns `Spliterator<T>`. If only `Iterable<T>#spliterator()` returned `Spliterator<? extends T>` instead of `Spliterator<T>` 🤦‍♂️

Reply:
fair enough - use the cast

Reply:
Done